api/codelists.go

```diff
@@ -14,13 +15,32 @@ func (c *CodeListAPI) getCodeLists(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	filterBy := r.URL.Query().Get("type")
 
+	// get limit from query parameters, or default value
+	limit, err := GetPositiveIntQueryParameter(r.URL.Query(), "limit", c.defaultLimit)
+	if err != nil {
+		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR)
```

Comment:
I was going to suggest moving this log line into the `handleError` function, but then realised it's already logging the error. I think the log line above can be removed. It's probably worth replacing the `logMsg` that's passed to the `handleError` function with the above log message, as it has more detail. If you agree with the change - it can be applied to all places where the query string is parsed (two cases in each handler)